gui/wxpython/vdigit/toolbars.py

```diff
@@ -882,6 +946,43 @@ def StartEditing(self, mapLayer):
             else:
                 return
 
+        # check and clean zero length lines, boundaries
```

Comment:
This function is already long, but if we add all these lines with two new return statements and at least six branches, it will be only harder to maintain later. If you stay with this solution, I suggest refactoring or shortening the code needed right here.